libs/blockscout-service-launcher/src/env_collector.rs

```diff
@@ -0,0 +1,505 @@
+    skip_vars: Vec<String>,
+) -> Result<Vec<EnvVariable>, anyhow::Error>
+where
+    S: ConfigSettings + Serialize + DeserializeOwned,
+{
+    let example = Envs::from_example_toml::<S>(
+        service_name,
+        std::fs::read_to_string(config_path)
+            .context("failed to read example file")?
+            .as_str(),
+        skip_vars,
+    )?;
+    let markdown: Envs = Envs::from_markdown(
+        std::fs::read_to_string(markdown_path)
+            .context("failed to read markdown file")?
+            .as_str(),
+    )?;
+
+    let missing = example
+        .vars
+        .iter()
+        .filter(|(key, value)| {
+            let maybe_markdown_var = markdown.vars.get(*key);
+            maybe_markdown_var.map(|var| var != *value).unwrap_or(true)
+        })
+        .map(|(_, value)| value.clone())
+        .collect();
+
+    Ok(missing)
+}
+
+fn update_markdown_file<S>(
+    service_name: &str,
+    markdown_path: &Path,
+    config_path: &Path,
+    skip_vars: Vec<String>,
+) -> Result<(), anyhow::Error>
+where
+    S: ConfigSettings + Serialize + DeserializeOwned,
+{
+    let from_config = Envs::from_example_toml::<S>(
+        service_name,
+        std::fs::read_to_string(config_path)
+            .context("failed to read config file")?
+            .as_str(),
+        skip_vars,
+    )?;
+    let mut markdown_config = Envs::from_markdown(
+        std::fs::read_to_string(markdown_path)
+            .context("failed to read markdown file")?
+            .as_str(),
+    )?;
+    markdown_config.update_no_override(from_config);
+    let table = serialize_env_vars_to_md_table(markdown_config);
+
+    let content = std::fs::read_to_string(markdown_path).context("failed to read markdown file")?;
+    let lines = content.lines().collect::<Vec<&str>>();
+    let line_start = lines
+        .iter()
+        .position(|line| line.contains(ANCHOR_START))
+        .context("anchors.envs.start not found in markdown")?;
+    let line_end = lines
+        .iter()
+        .position(|line| line.contains(ANCHOR_END))
+        .context("anchors.envs.end not found in markdown")?;
+
+    let new_content = [&lines[..=line_start], &[&table], &lines[line_end..]].concat();
+    std::fs::write(markdown_path, new_content.join("\n")).context("failed to write file")?;
+    Ok(())
+}
+
+fn var_is_required<S>(settings: &S, path: &str) -> bool
+where
+    S: ConfigSettings + Serialize + DeserializeOwned,
+{
+    let mut json = serde_json::to_value(settings).unwrap();
+    json.dot_remove(path).unwrap();
+    let result = serde_json::from_value::<S>(json);
+    result.is_err()
+}
+
+fn from_key_to_json_path(key: &str, service_prefix: &str) -> String {
+    key.trim_start_matches(&format!("{service_prefix}__"))
+        .to_lowercase()
+        .replace("__", ".")
+        .to_string()
+}
+
+fn serialize_env_vars_to_md_table(vars: Envs) -> String {
+    let mut result = r#"
+| Variable | Is required | Example value | Comment |
```

Comment:
Do you think it make sense to stick to the same format as in Blockscout docs - https://github.com/blockscout/docs/blob/master/for-developers/information-and-settings/env-variables.md?
```suggestion
| Variable | Required | Description | Default value |
```

Reply:
sure, will fix it